### PR TITLE
Fix error with ARNID lookup

### PR DESCRIPTION
**Pull request by akhalina**
I made the mistake in my previous PR of not properly testing the ARNID lookup. I needed to use pointers properly for that one as well. This time I tested the ip, hostname (these use the same method), and arnid lookups locally.

### CODE REVIEW

File: pkg/storage/db.go [hunk 1]
```diff
@@ -864,11 +864,13 @@ func (db *DB) FetchByARNID(ctx context.Context, when time.Time, arnID string) ([
 			hasTag = true
 		}
 
-		tempChampionsMap[chLogin+chEmail] = domain.Person{
-			Login: &chLogin,
-			Email: &chEmail,
-			Name:  &chName,
-			Valid: &chValid,
+		if chLogin != nil {
+			tempChampionsMap[*chLogin] = domain.Person{
```
Comment (akhalina) [inline pkg/storage/db.go:868]: I'm not 100% sure about this change. Apparently the login is unique, and I assume the same champion would not appear for the same resource twice. But please correct me if I'm wrong.

Comment (gcase555) [reply pkg/storage/db.go:868]: It's technically possible an account owner could also be an account champion although I don't know of any cases in our data where this is true.

Comment (akhalina) [reply pkg/storage/db.go:868]: In this case the old way this was done would still cause issues correct?

Comment (gcase555) [reply pkg/storage/db.go:868]: Cause what specific issues?

Comment (gcase555) [reply pkg/storage/db.go:868]: I am looking at how the IP based queries are executed and the logic is slightly different which is interesting, is it possible the issue exists there as well?

Comment (akhalina) [reply pkg/storage/db.go:868]: I guess I just want to know why we had the map key as chLogin+chEmail before? Is there some risk to just using the login?

Comment (akhalina) [reply pkg/storage/db.go:868]: Sorry, to clarify there was no issue that I know of in the previous code. I can add a check for nil on both the login and email, I just don't know if that is necessary/preferred.

Comment (gcase555) [reply pkg/storage/db.go:868]: Login is guaranteed to be unique so I am unsure why they would combine the two for a key also

**Pull request merged**